v-next/hardhat/src/internal/cli/main.ts

```diff
@@ -98,7 +98,7 @@ export async function main(
       return;
     }
 
-    if (options.registerTsx) {
+    if (options.registerTsx === true) {
```

Comment:
we need `=== true` to satisfy the linter (because it complains `if (options.registerTsx)` is not clear if this is meant as a non-null check or as a check that the variable is true)